### ISSUE TITLE
train.py

### ISSUE THREAD

**Issue by Viky397** [BerkeleyAutomation/sd-maskrcnn#51]
Hello,

When I ` python3 tools/train.py`, it is frozen at Epoch 1/20, and then the process `Killed`.
Any suggestions? Thank you in advance!

**Comment by mjd3** [comment]
Possibly an OOM error? Or running out of RAM? It is hard to say without more information, but hopefully that helps!

**Comment by Viky397** [comment]
Yes it's probably a RAM error, Thank you! 
Another quick question, does the code save the training losses anywhere? I'd be interested in doing a Bias-Variance tradeoff study.

**Comment by mjd3** [comment]
Yup! By default, it logs to TensorBoard (see [here](https://github.com/BerkeleyAutomation/Mask_RCNN/blob/351ea1826f50cdc3025dcd7eeebd62a2ea4fa5dc/mrcnn/model.py#L2341)).

**Comment by Viky397** [comment]
Thank you! I'm interested in further fine tuning the pre-trained model. As I don't have access to a very powerful GPU, I won't be able to train it from scratch. I was thinking I could perhaps tweak the learning rate, but is there any way for me to perform this fine-tuning without having to retrain the model from scratch?
Thank you!

**Comment by Viky397** [comment]
As well, is there any way for me to access the losses without having to retrain?

**Comment by mjd3** [comment]
Yes; you can finetune by pointing to weights that already exist in the `train.yaml` file (see [here](https://github.com/BerkeleyAutomation/sd-maskrcnn/blob/75f2a81108c9e74702f831f60fcef8c5e903c8ab/cfg/train.yaml#L3) and [here](https://github.com/BerkeleyAutomation/sd-maskrcnn/blob/75f2a81108c9e74702f831f60fcef8c5e903c8ab/tools/train.py#L97) for details). This is also how you would finetune starting from COCO or ImageNet weights. There is currently no way to access training losses from the pretrained model; sorry about that.

**Comment by Viky397** [comment]
No worries, Thank you very much! Curious- how does WISDOM differ from COCO standards?

**Comment by mjd3** [comment]
I'm not sure I'm completely understanding the question, but if you're asking what the main differences are between the COCO dataset and WISDOM dataset, [this page](https://sites.google.com/view/wisdom-dataset/dataset_links) summarizes it well (it is also included in the supplement for the paper). Let me know if that answers your question!

**Comment by Viky397** [comment]
Yes, perfect, thank you so much! :)

**Comment by Viky397** [comment]
So sorry, one more question:

**Comment by mjd3** [comment]
This is just meant to be a normalization step to minimize some of the effects of the actual distance of the camera - we subtract off the mean pixel value from each of the input images. Thus, if you have images that are at a different depth than our setup, there may be more generalization capability.

**Comment by Viky397** [comment]
Thank you!

**Comment by Viky397** [comment]
Sorry, another questions! When you say "We reduce the number of classes to two. Each proposed instance mask is classified as background or as a foreground object. Of these, only foreground masks are visualized."

Was this done by introducing an attention mechanism into Mask RCNN? Thank you!
